File: app/src/main/java/com/squareoverlay/NextLineButtonView.java
package com.squareoverlay;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;

public class NextLineButtonView extends BaseButtonView {

    private static final int BUTTON_WIDTH = 280;
    private static final int BUTTON_HEIGHT = 200;
    private static final int CORNER_RADIUS = 30;

    public NextLineButtonView(Context context) {
        super(context, BUTTON_WIDTH, BUTTON_HEIGHT, CORNER_RADIUS);
        buttonTextPaint.setTextSize(50);
    }

    @Override
    protected int getGradientStartColor() {
        return Color.rgb(33, 150, 243); // Light blue
    }

    @Override
    protected int getGradientEndColor() {
        return Color.rgb(0, 120, 200); // Darker blue
    }

    @Override
    protected int getPressedColor() {
        return Color.rgb(0, 100, 150); // Darker blue when pressed
    }

    @Override
    protected void drawButtonContent(Canvas canvas) {
        float textY = buttonRect.centerY() + (buttonTextPaint.getTextSize() / 2) - 10;
        canvas.drawText("↓", buttonRect.centerX(), textY, buttonTextPaint);
    }
}

File: app/src/main/java/com/squareoverlay/NextLineButtonView.kt
package com.squareoverlay

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color

class NextLineButtonView(context: Context?) :
    BaseButtonView(context, BUTTON_WIDTH, BUTTON_HEIGHT, CORNER_RADIUS) {

    init {
        buttonTextPaint.textSize = 50f
    }

    override val gradientStartColor: Int
        get() = Color.rgb(33, 150, 243) // Light blue

    override val gradientEndColor: Int
        get() = Color.rgb(0, 120, 200) // Darker blue

    override val pressedColor: Int
        get() = Color.rgb(0, 100, 150) // Darker blue when pressed

    override fun drawButtonContent(canvas: Canvas?) {
        canvas?.let {
            val textY = buttonRect.centerY() + (buttonTextPaint.textSize / 2) - 10
            it.drawText("↓", buttonRect.centerX(), textY, buttonTextPaint)
        }
    }

    companion object {
        private const val BUTTON_WIDTH = 280
        private const val BUTTON_HEIGHT = 200
        private const val CORNER_RADIUS = 30
    }
}
